src/utils/gcsApi.ts

```diff
@@ -172,24 +173,30 @@ export class CraftGCSClient {
   }
 
   /**
-   * Uploads the artifact at the given local path to the path specified in the
-   * given config object
+   * Uploads the given artifact to the path specified in the given config object.
    *
-   * @param localFilePath Location of the file to be uploaded
+   * @param artifact The artifact to be uploaded
    * @param uploadConfig Configuration for the upload including destination path
    * and metadata
    */
-  private async uploadArtifactFromPath(
-    localFilePath: string,
+  private async uploadArtifact(
+    artifact: CraftArtifact,
```

Comment:
Does this mean we have to provided the required "storedFile" attribute in `artifact` now?

Reply:
Shoot. Good point. Not a problem for the target, but would be a problem for the artifact provider, before the files have been uploaded. Thanks for catching that. Will fix.

Reply:
Haven't checked in details, but probably we can just make `storedFile` optional.